python/mooseutils/ReporterReader.py

```diff
@@ -43,13 +44,20 @@ class ReporterReader(object):
         print('value_name in object object_name was NOT found!')
     ```
     """
-    def __init__(self, file):
+    def __init__(self, file, part=None):
         self._filename = file
         self._data = dict(time_steps=[])
         self._time = -1
         self._index = 0
         self.update()
 
+        # WIP ADD TEST
+        if part is not None and part != 0:
+            if part >= self.numParts():
+                raise Exception('Cannot load part {}; "{}" has {} parts'.format(part, file, self.numParts()))
+            self._filename = file + '.' + '{}'.format(part).zfill(len(str(self.numParts())))
```

Comment:
I went back and forth on this before, I decided it wasn't a big deal since most stuff is usually reported on root, but with this I feel it's a good idea to support distributed reporters. Therefore, I think it would be a good idea to support globs for the distributed filenames. With this, the `update()` function should probably have a `part` parameter, which will search for the globbed files. Unfortunately, there could be two globbed parts of the filename for `one_file_per_timestep` parameter, i.e. `file_*.json*`. This may be a little tricky to handle. But I will leave it up to you to handle.

Reply:
Good point on distributed reporters.  It would be great to be able to get this info back from all procs.